src/Microsoft.Azure.WebJobs.Protocols/FunctionDescriptor.cs

```diff
@@ -35,5 +36,31 @@ public class FunctionDescriptor
         /// </summary>
         [JsonIgnore]
         internal MethodInfo Method { get; set; }
+
+        /// <summary>
+        /// Gets the <see cref="TriggerParameterDescriptor"/> for this function
+        /// </summary>
+        [JsonIgnore]
+        internal TriggerParameterDescriptor TriggerDescriptor { get; set; }
+#if PUBLICPROTOCOL
+#else
+        /// <summary>
+        /// Gets the <see cref="TraceLevel"/> for this function
+        /// </summary>
+        [JsonIgnore]
+        internal TraceLevel LogLevel { get; set; }
```

Comment:
I'd call all of these new properties the same name as their type (`TraceLevel`, `TimeoutAttribute`, `SingletonAttributes`, etc). It's more descriptive of what they actually are. Also, `ILogger` has a `LogLevel`, so changing this to `TraceLevel` removes some future confusion.